design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,126 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low (limited growth is still acceptable)
+
+## Metrics for Rook to collect
+Metric names will indicate a hierarchy that can be parsed to add it to Ceph telemetry collection in
+a more ordered fashion.
+
+For example `rook/version` and `rook/kubernetes/version` would be put into a structure like shown:
+```json
+"rook": {
+  "version": "vx.y.z"
+  "kubernetes": {
+    "version": "vX.Y.Z"
+  }
+}
+```
+
+### Overall metrics
+- `rook/version` - Rook version.
+- `rook/kubernetes/...`
+  - `rook/kubernetes/version` - Kubernetes version.
+    - Ceph already collects os/kernel versions.
+  - `rook/kubernetes/node/count/total` - Number of Kubernetes nodes.
+- `rook/csi/...`
+  - `rook/csi/version` - Ceph CSI version.
+  - `rook/csi/plugin-pod/...` - Info about CSI plugin pods
+    - `rook/csi/plugin-pod/count/...` - Info about how many CSI plugin pods exist and of what types
+      This gives idea of how many k8s nodes are in the k8s cluster that are allowed to use Ceph
+      storage. With different types, telemetry can show if some CSI drivers are ever disabled.
+      - `rook/csi/plugin-pod/count/rbd`
+      - `rook/csi/plugin-pod/count/cephfs`
+      - `rook/csi/plugin-pod/count/nfs`
+
+### CephCluster metrics
+- `rook/cluster/node/count/with-ceph-pods` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we will report `-1` to represent "unknown"
+- `rook/cluster/storage-class/...` - Info about storage classes related to the current Ceph cluster
```

Comment:
Since everything is part of the same cluster, how about if we remove "cluster" from the path? This simplifies the path for most of the metrics.
```suggestion
- `rook/storage-class/...` - Info about storage classes related to the current Ceph cluster
```